### PR TITLE
feat: fetch round measurements from on-chain data in dry-run

### PR DESCRIPTION
**Pull request by bajtos**
Simplify debugging of historic SPARK rounds.

### CODE REVIEW

File: bin/dry-run.js [hunk 1]
```diff
@@ -26,6 +31,35 @@ if (!roundIndex) {
...
+    ),
+    provider
+  )
+
+  console.log('Fetching MeasurementsAdded events from the ledger')
+  // TODO: filter only measurements for the given `roundIndex`
+  // See https://github.com/Meridian-IE/impact-evaluator/issues/57
+  // const filter = ieContract.filters.MeasurementsAdded()
+  const filter = { address: ieContract.address }
+  console.log(filter)
```
Comment (bajtos) [inline bin/dry-run.js:52]:
```json
{ address: '0xeeadb614b63dee83f0e7b4095094ae7c5d439ba2' }
```

File: bin/dry-run.js [hunk 2]
```diff
@@ -26,6 +31,35 @@ if (!roundIndex) {
...
+
+  const logs = await provider.getLogs({ ...filter, fromBlock: -1900 })
+  console.log('logs', logs)
```
Comment (bajtos) [inline bin/dry-run.js:55]:
```js
{
  action: 'request',
  request: {
    method: 'eth_getLogs',
    params: [ [Object] ],
    id: 46,
    jsonrpc: '2.0'
  },
  provider: JsonRpcProvider {
    _isProvider: true,
    _events: [ [Event] ],
    _emitted: { block: -2 },
    disableCcipRead: false,
    formatter: Formatter { formats: [Object] },
    anyNetwork: false,
    _networkPromise: Promise { [Object] },
    _maxInternalBlockNumber: 3414197,
    _lastBlockNumber: -2,
    _maxFilterBlockRange: 10,
    _pollingInterval: 4000,
    _fastQueryDate: 1700732343994,
    connection: { url: 'https://api.node.glif.io/rpc/v0' },
    _nextId: 47,
    _eventLoopCache: { detectNetwork: null, eth_chainId: null, eth_blockNumber: null },
    _network: { chainId: 314, name: 'unknown' },
    _internalBlockNumber: Promise { [Object] },
    _fastBlockNumber: 3414197,
    _fastBlockNumberPromise: Promise { 3414197 }
  }
}
logs []
```

File: bin/dry-run.js [hunk 3]
```diff
@@ -26,6 +31,35 @@ if (!roundIndex) {
...
+  const events = await ieContract.queryFilter(
+    filter,
+    // max lookback period allowed by Glif.io, approx 16h40m
+    -1998
+  )
+  console.log('events', events)
```
Comment (bajtos) [inline bin/dry-run.js:61]:
```js
{
  action: 'request',
  request: {
    method: 'eth_getLogs',
    params: [ [Object] ],
    id: 48,
    jsonrpc: '2.0'
  },
  provider: JsonRpcProvider {
    _isProvider: true,
    _events: [ [Event] ],
    _emitted: { block: -2 },
    disableCcipRead: false,
    formatter: Formatter { formats: [Object] },
    anyNetwork: false,
    _networkPromise: Promise { [Object] },
    _maxInternalBlockNumber: 3414203,
    _lastBlockNumber: -2,
    _maxFilterBlockRange: 10,
    _pollingInterval: 4000,
    _fastQueryDate: 1700732544523,
    connection: { url: 'https://api.node.glif.io/rpc/v0' },
    _nextId: 49,
    _eventLoopCache: { detectNetwork: null, eth_chainId: null, eth_blockNumber: null },
    _network: { chainId: 314, name: 'unknown' },
    _internalBlockNumber: Promise { [Object] },
    _fastBlockNumber: 3414203,
    _fastBlockNumberPromise: Promise { 3414203 }
  }
}
events []
```

**Comment by bajtos** [comment]
Example test case:

```
DEBUG=spark:* node bin/dry-run.js 0xeeadb614b63dee83f0e7b4095094ae7c5d439ba2 428
```

It prints the messages I posted above. No events were found.

**Comment by bajtos** [comment]
I fixed the problem by calling JSON RPC API manually via `fetch` 🤷🏻 

I verified the implementation on a dry run for round 526.

**Review by juliangruber — APPROVED** [review]

**Pull request merged**